crates/bevy_transform/src/components/transform.rs

```diff
@@ -325,17 +333,28 @@ impl Transform {
 
     /// Rotates this [`Transform`] so that [`Transform::forward`] points towards the `target` position,
     /// and [`Transform::up`] points towards `up`.
+    ///
+    /// It is not possible to construct a rotation when the resulting forward direction is parallel with `up`.
+    /// If this happens, an orthogonal vector to `up` will be used as the "right" direction to get a valid value.
+    /// It is also not possible when the target is at the same place as the transform. In this case, the Z axis will be used as the "forward" direction.
     #[inline]
     pub fn look_at(&mut self, target: Vec3, up: Vec3) {
         self.look_to(target - self.translation, up);
     }
 
     /// Rotates this [`Transform`] so that [`Transform::forward`] points in the given `direction`
     /// and [`Transform::up`] points towards `up`.
+    ///
+    /// It is not possible to construct a rotation when `direction` is parallel with `up`.
+    /// If this happens, an orthogonal vector to `up` will be used as the "right" direction to get a valid value.
+    /// It is also not possible when the direction provided is zero. In this case, the Z axis will be used as the "forward" direction.
     #[inline]
     pub fn look_to(&mut self, direction: Vec3, up: Vec3) {
-        let forward = -direction.normalize();
-        let right = up.cross(forward).normalize();
+        let forward = -direction.try_normalize().unwrap_or(Vec3::Z);
+        let right = up
+            .cross(forward)
+            .try_normalize()
+            .unwrap_or_else(|| up.any_orthogonal_vector());
```

Comment:
```suggestion
            .unwrap_or_else(|| up.any_orthonormal_vector());
```
`any_orthogonal_vector()` might not return a normalized vector.

Also, should we check whether `up` is normalized and non-zero as well?

Reply:
`any_orthonormal_vector` only does if `up` is normalized, otherwise it returns garbage

Reply:
Looks like the docs for `any_orthonormal_vector` are wrong then